src/js/media.js

```diff
@@ -69,119 +70,142 @@ import { __ } from '@wordpress/i18n';
 										'attachment-details-description'
 									);
 
-								if ( textField ) {
+								if (textField) {
 									textField.value = resp;
 								}
 							}
 						}
 					},
-				} )
+				})
 			);
 		}
 
-		if ( imageTagsButton ) {
-			imageTagsButton.addEventListener( 'click', ( e ) =>
-				handleClick( {
+		if (imageTagsButton) {
+			imageTagsButton.addEventListener('click', (e) =>
+				handleClick({
 					button: e.target,
 					endpoint: '/classifai/v1/image-tags/',
-				} )
+				})
 			);
 		}
 
-		if ( ocrScanButton ) {
-			ocrScanButton.addEventListener( 'click', ( e ) =>
-				handleClick( {
+		if (ocrScanButton) {
+			ocrScanButton.addEventListener('click', (e) =>
+				handleClick({
 					button: e.target,
 					endpoint: '/classifai/v1/ocr/',
-					callback: ( resp ) => {
-						if ( resp ) {
+					callback: (resp) => {
+						if (resp) {
 							const textField =
 								document.getElementById(
 									'attachment-details-two-column-description'
 								) ??
 								document.getElementById(
 									'attachment-details-description'
 								);
-							if ( textField ) {
+							if (textField) {
 								textField.value = resp;
 							}
 						}
 					},
-				} )
+				})
 			);
 		}
 
-		if ( smartCropButton ) {
-			smartCropButton.addEventListener( 'click', ( e ) =>
-				handleClick( {
+		if (smartCropButton) {
+			smartCropButton.addEventListener('click', (e) =>
+				handleClick({
 					button: e.target,
 					endpoint: '/classifai/v1/smart-crop/',
-				} )
+				})
 			);
 		}
 
-		if ( readButton ) {
-			readButton.addEventListener( 'click', ( e ) => {
-				const postID = e.target.getAttribute( 'data-id' );
-				wp.apiRequest( { path: `/classifai/v1/read-pdf/${ postID }` } );
-				e.target.setAttribute( 'disabled', 'disabled' );
-				e.target.textContent = __( 'Read API requested!', 'classifai' );
-			} );
+		if (readButton) {
+			readButton.addEventListener('click', (e) => {
+				const postID = e.target.getAttribute('data-id');
+				wp.apiRequest({ path: `/classifai/v1/read-pdf/${postID}` });
+				e.target.setAttribute('disabled', 'disabled');
+				e.target.textContent = __('Read API requested!', 'classifai');
+			});
+		}
+
+		if (transcribeButton) {
+			transcribeButton.addEventListener('click', (e) =>
+				handleClick({
+					button: e.target,
+					endpoint: '/classifai/v1/openai/generate-transcript/',
+					callback: (resp) => {
+						if (resp) {
+							const textField =
+								document.getElementById(
+									'attachment-details-two-column-description'
+								) ??
+								document.getElementById(
+									'attachment-details-description'
+								);
+							if (textField) {
+								textField.value = resp;
+							}
+						}
+					},
```

Comment:
```suggestion
					},
					buttonText: __('Re-transcribe', 'classifai'),
```
To keep the button text "Re-transcribe" after the process is complete. Currently, it becomes "Rescan".